redwood-tooling-codegen/src/main/kotlin/app/cash/redwood/tooling/codegen/testingGeneration.kt

```diff
@@ -278,24 +279,53 @@ internal fun generateMutableWidget(schema: Schema, widget: Widget): FileSpec {
 
 /*
 public class ButtonValue(
-  public override val modifier: Modifier = Modifier,
-  public val text: String? = null,
-  public val enabled: Boolean = false,
+  override val modifier: Modifier = Modifier,
+  public val text: String?,
+  public val onClick: (() -> Unit)?,
 ) : WidgetValue {
-  public override val childrenLists: List<List<WidgetValue>>
+  override val childrenLists: List<List<WidgetValue>>
     get() = listOf()
 
-  public override fun equals(other: Any?): Boolean = other is SunspotButtonValue &&
+  override fun equals(other: Any?): Boolean = other is SunspotButtonValue &&
     other.text == text &&
     other.enabled == enabled
 
-  public override fun hashCode(): Int = listOf(
+  override fun hashCode(): Int = listOf(
     text,
     enabled,
   ).hashCode()
 
-  public override fun toString(): String =
+  override fun toString(): String =
     """ButtonValue(text=$text, enabled=$enabled)"""
+
+  override fun toDebugString(): String {
+    val debugString = """
+        |ButtonValue(
+        |  text = $text
+        |)
+        """.trimMargin()
+    val childrenDebugString = childrenLists.joinToString(prefix = " {\n", postfix = "\n}", separator
+        = "\n") {
+      it.toDebugString().prependIndent("  ")
+    }
```

Comment:
I'm having trouble picturing how a multi-children widget renders. It looks like maybe all of the children are put into a single set of curly braces? Meaning you wouldn't be able to differentiate between them.

You might consider making widgets with multiple children render them more like properties.
```
TestRowValue {  // Single children, renders with trailing lambda-like syntax
  MultiChildrenValue(
    front = {
      Button(..)
      Button(..)
    }
    back = {
      Text(..)
      Text(..)
    }
}
```
of course we could render all nodes like this, but it introduces an additional layer of nesting for the single-children case which is a bit annoying.

Reply:
Yep, currently they would all be in one set of braces. Will try to set it up like above 👍